src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -447,6 +442,15 @@ bool SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets, u
         is_new = true;
       }
 
+      if (primary_ecu_serial == ecu_serial) {
+        if (!target.IsOstree() &&
+            (config.pacman.type == PackageManager::kOstree || config.pacman.type == PackageManager::kOstreeDockerApp)) {
+          LOG_ERROR << "Cannot install a non-OSTree package on an OSTree system";
+          last_exception = Uptane::InvalidMetadata(target.filename(), "", "Non-OSTree package");
```

Comment:
That's not a bad choice, but since it's usually used for problems in parsing the metadata, maybe we should find something else. Nothing else looks better, though, so how about making a new exception? `InvalidTarget` perhaps?

Reply:
Sure，I was consider making a new one too. Thanks!